### PR TITLE
Add smoketest MVP

### PR DESCRIPTION
**Pull request by gearama**
Added smoketest that pulls the keyvault sdks from vcpkg and should run it along side the other samples. 
The smoketest creates one of each clients and calls a single api from each sdk to ensure all is working properly.
#closes 3035

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: samples/integration/vcpkg-all-smoke/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+add_executable (
+  smoketest-vcpkg
+    src/main
+)
+
+# Link to Azure SDK
+target_link_libraries(smoketest-vcpkg
+     PRIVATE
+     Azure::azure-security-keyvault-keys
```
[the remark this answers is not in the document]
Comment (gearama) [reply samples/integration/vcpkg-all-smoke/CMakeLists.txt:27]: is it published to vcpkg ?

[the remark this answers is not in the document]
Comment (danieljurek) [reply samples/integration/vcpkg-all-smoke/CMakeLists.txt:27]: Attestation is published to our daily branch on vcpkg. -- https://github.com/microsoft/vcpkg/pull/17119/files 

One can clone the `azure-sdk:release/azure-sdk-for-cpp-daily-verification` branch from `azure-sdk/vcpkg` to get the latest set of packages... 

However, if some packages are not published to vcpkg and fail CI (and don't get published to the daily vcpkg branch) then the Smoke Tests will fail. The failure would be expected in that situation and the fix would involve starting with fixing CI such that the packages themselves publish.

**Review by danieljurek — APPROVED** [review]
This moves us forward toward a smoke testing strategy as outlined here: https://gist.github.com/ckairen/f638231d6d146fef7b73f35d6ca7f7e0#pipeline-design

At the maximalist end of the spectrum we would deploy resources and test e2e scenarios but that's not in the explicit scope early on. 

The additional coverage in the samples test step is not required but nice to have because it could catch problems before merge. 


There will be future work items for integrating with the smoke test engineering system as that comes together (see link above)

**Pull request merged**